### PR TITLE
Add HTTP/2 support

### PR DESCRIPTION
**Pull request by kinetifex**
This adds the ability to set an `http2` config property for setting up HTTP/2 servers.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -81,14 +87,16 @@ module.exports = async function createServers(options, listening) {
 function normalizeOptions(options) {
   const http = normalizeHttpOptions(options.http, options);
   const https = normalizeHttpsOptions(options.https, options);
+  const http2 = normalizeHttpsOptions(options.http2, options);
 
-  if (!http && !https) {
+  if (!http && !https && !http2) {
     throw new Error('http and/or https are required options');
```
Comment (jpage-godaddy) [inline index.js:93]: Should this now read "http, https, and/or http2" are required options?

File: README.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ a node-style callback. The config object must have at minimum an `http` or
...
+| `http2`                  | Optional object. If present, an HTTP/2 server is started. You may start multiple HTTP/2 servers by passing an array of objects                                                                                                                                                        |
+| `http2.allowHTTP1`       | Enable [ALPN negotiation] allowing support for both HTTPS and HTTP/2 on the same socket.                                                                                                                                                                                              |
+| `http2.*`                | Any other properties supported by [http2.createSecureServer](https://nodejs.org/dist/latest-v8.x/docs/api/http2.html#http2_http2_createsecureserver_options_onrequesthandler).                                                                                                        |
```
Comment (jpage-godaddy) [inline README.md:35]: Should this also mention all the same certificate/secure options from `https` are allowed?

**Review by jpage-godaddy — APPROVED** [review]

File: index.js [hunk 3]
```diff
@@ -350,7 +358,11 @@ async function createHttps(ssl, log) {
     }
 
     log('https | listening on %d', port);
-    server = https.createServer(finalHttpsOptions, ssl.handler);
+    if(h2) {
+      server = require('http2').createSecureServer(finalHttpsOptions, ssl.handler)
+    } else {
+      server = https.createServer(finalHttpsOptions, ssl.handler);
```
Comment (mswaagman-godaddy) [inline index.js:364]: Should this now also 'lazily' require `https`? If both lines are consistent you could 

```suggestion
      server = require(h2 ? 'http2' : 'https').createServer(finalHttpsOptions, ssl.handler);
```

**Review by mswaagman-godaddy — APPROVED** [review]
Non blocking comment

File: index.js (hunk 3 above)
Comment (kinetifex) [reply index.js:364]: We could late http2 uses `.createSecureServer`, but still, yes we could lazy require `https` (and `http` for that matter) which would clean up the variable shadowing that is occurring. — Speaking of which, this repo could benefit from eslint. I may do a follow-up PR with that.

**Pull request merged**